brat/Brat/Checker.hs

```diff
@@ -122,45 +122,41 @@ checkWire Kerny (WC fc tm) outputs (dangling, ot) (hungry, ut) = localFC fc $ do
     else typeEq (show tm) (Dollar []) ut ot
   wire (dangling, ot, hungry)
 
-checkInputs :: (CheckConstraints m KVerb, ?my :: Modey m)
+checkIO :: forall m d k exp act . (CheckConstraints m k, ?my :: Modey m)
+        => WC (Term d k)
+        -> [(NamedPort exp, BinderType m)]
+        -> [(NamedPort act, BinderType m)]
+        -> ((NamedPort exp, BinderType m) -> (NamedPort act, BinderType m) -> Checking ())
+        -> String
+        -> Checking [(NamedPort exp, BinderType m)] -- left(overs/unders)
+checkIO tm@(WC fc _) exps acts wireFn errMsg = do
+  let _ = ?my -- otherwise ?my is "redundant" but typechecking fails without it
```

Comment:
I tried `EvMode m` and `MODEY m` instead of `?my :: Modey m` in the signature but no luck, they failed with various `m0 != m` errors noting that "BinderType is a non-injective type family"

Reply:
It might work with a `modily ?my $ do ...`

Reply:
Yes! Thanks :-)